controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

joint_state_broadcaster:
  type: joint_state_broadcaster/JointStateBroadcaster

joint_trajectory_controller:
  type: joint_trajectory_controller/JointTrajectoryController
  joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - joint6
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    joint1: {trajectory: 0.1, goal: 0.1}
    joint2: {trajectory: 0.1, goal: 0.1}
    joint3: {trajectory: 0.1, goal: 0.1}
    joint4: {trajectory: 0.1, goal: 0.1}
    joint5: {trajectory: 0.1, goal: 0.1}
    joint6: {trajectory: 0.1, goal: 0.1}
  state_publish_rate:  50
  action_monitor_rate: 20

# 可选：位置控制器
position_controllers:
  type: position_controllers/JointGroupPositionController
  joints:
    - joint1
    - joint2
    - joint3
    - joint4
    - joint5
    - joint6